lib/galaxy/workflow/modules.py

```diff
@@ -980,7 +980,7 @@ def get_runtime_inputs(self, connections=None, **kwds):
                     tool_inputs = module.tool.inputs  # may not be set, but we're catching the Exception below.
 
                     def callback(input, prefixed_name, context, **kwargs):
-                        if prefixed_name == connection.input_name:
+                        if prefixed_name == connection.input_name and not input.type == "text":
```

Comment:
I think I would be more comfortable with ``hasattr(get_options)`` - seems like it would be less brittle against adding/removing parameter types?

Reply:
Just looking through the code, all parameter types that have `get_options` are all subclasses of `SelectToolParameter`. Given this was added specifically for select tool parameter, should we make this
```suggestion
                        if prefixed_name == connection.input_name and isinstance(input, SelectToolParameter):
```
instead ? This seems a little clearer in intent ?

Reply:
Either is fine with me - I mildly prefer hasattr as being a little more Pythonic IMO.

Reply:
Got it, thanks @jmchilton!